tests/operators/fermi_hubbard_test.py

```diff
@@ -0,0 +1,165 @@
+    potential."""
+
+    # open boundary conditions
+    op = fermi_hubbard_1d(norb=4, tunneling=1, interaction=2, chemical_potential=3)
+    ham = ffsim.linear_operator(op, norb=4, nelec=(2, 2))
+    eigs, _ = scipy.sparse.linalg.eigsh(ham, which="SA", k=1)
+    np.testing.assert_allclose(eigs[0], -14.875942809005)
+
+    # periodic boundary conditions
+    op_periodic = fermi_hubbard_1d(
+        norb=4, tunneling=1, interaction=2, chemical_potential=3, periodic=True
+    )
+    ham_periodic = ffsim.linear_operator(op_periodic, norb=4, nelec=(2, 2))
+    eigs_periodic, _ = scipy.sparse.linalg.eigsh(ham_periodic, which="SA", k=1)
+    np.testing.assert_allclose(eigs_periodic[0], -14.828427124746)
+
+
+def test_fermi_hubbard_1d_with_nearest_neighbor_interaction():
+    """Test 1D Fermi-Hubbard model Hamiltonian with onsite interaction, chemical
+    potential, and nearest-neighbor interaction."""
+
+    # open boundary conditions
+    op = fermi_hubbard_1d(
+        norb=4,
+        tunneling=1,
+        interaction=2,
+        chemical_potential=3,
+        nearest_neighbor_interaction=4,
+    )
+    ham = ffsim.linear_operator(op, norb=4, nelec=(2, 2))
+    eigs, _ = scipy.sparse.linalg.eigsh(ham, which="SA", k=1)
+    np.testing.assert_allclose(eigs[0], -9.961978205599)
+
+    # periodic boundary conditions
+    op_periodic = fermi_hubbard_1d(
+        norb=4,
+        tunneling=1,
+        interaction=2,
+        chemical_potential=3,
+        nearest_neighbor_interaction=4,
+        periodic=True,
+    )
+    ham_periodic = ffsim.linear_operator(op_periodic, norb=4, nelec=(2, 2))
+    eigs_periodic, _ = scipy.sparse.linalg.eigsh(ham_periodic, which="SA", k=1)
+    np.testing.assert_allclose(eigs_periodic[0], -8.781962448006)
+
+
+def test_fermi_hubbard_1d_with_unequal_filling():
+    """Test 1D Fermi-Hubbard model Hamiltonian with unequal filling."""
+
+    # open boundary conditions
+    op = fermi_hubbard_1d(
+        norb=4,
+        tunneling=1,
+        interaction=2,
+        chemical_potential=3,
+        nearest_neighbor_interaction=4,
+    )
+    ham = ffsim.linear_operator(op, norb=4, nelec=(1, 3))
+    eigs, _ = scipy.sparse.linalg.eigsh(ham, which="SA", k=1)
+    np.testing.assert_allclose(eigs[0], -6.615276287167)
+
+    # periodic boundary conditions
+    op_periodic = fermi_hubbard_1d(
+        norb=4,
+        tunneling=1,
+        interaction=2,
+        chemical_potential=3,
+        nearest_neighbor_interaction=4,
+        periodic=True,
+    )
+    ham_periodic = ffsim.linear_operator(op_periodic, norb=4, nelec=(1, 3))
+    eigs_periodic, _ = scipy.sparse.linalg.eigsh(ham_periodic, which="SA", k=1)
+    np.testing.assert_allclose(eigs_periodic[0], -0.828427124746)
+
+
+def test_fermi_hubbard_1d_hermiticity():
+    """Test 1D Fermi-Hubbard model Hamiltonian hermiticity."""
+
+    n_orbitals, n_alpha, n_beta = 4, 3, 1
+    dim = comb(n_orbitals, n_alpha) * comb(n_orbitals, n_beta)
```

Comment:
```suggestion
    norb = 4
    nelec = (3, 1)
    dim = ffsim.dim(norb, nelec)
```
Sorry for the extreme nitpicking but this is how it's done in the rest of the codebase.